src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionService.kt

```diff
@@ -41,7 +42,7 @@ class IdentificationSessionService(
     }
 
     private fun buildTcTokenUrl(session: IdentificationSession): String {
-        return "${applicationProperties.baseUrl}$TCTOKENS_BASE_PATH/${session.useIdSessionId}"
+        return "${applicationProperties.baseUrl}$TCTOKENS_BASE_PATH/${session.useIdSessionId}?$PARAM_NAME_TENANT_ID=${session.tenantId}"
```

Comment:
By appending the `tenant_id` to the tc token URL we are externalizing a value which is meant for internal purposes and tracking. I think we should find another solution to track the tenant for the tc token request. The tenant could for example be derived from the `useIdSessionId` which is included in the request. The `ResolveTenantFilter` could do that for example. Please reconsider this.

Reply:
Urs and I thought about this as well. By selecting the tenant via `useIdSessionId` within the `RequestMetricsTenantTagProvider`, we need to inject the repository and make another database call. 
However, maybe we find a way to attach the `tenantId` to the request within the `TcTokenController`. 
Prometheus uses two calls to track a route. A beginning call (start time), then logic, then ending call (stop time). Maybe we can read the attached `tenantId` at the ending call. Will try out.

Reply:
This is better. Could you maybe extract that into a method which says what it does so it's better understandable later?
And isn't that too late to be included in the tag for the metric? Isn't the metric already sent? 

Some more information about an additional database call: 
[Hibernate (ORM used by Spring data) caches all session objects in a "first level cache"](https://www.digitalocean.com/community/tutorials/hibernate-caching-first-level-cache). So requesting the identification session twice during one requests only results in one database query since afterwards the entity is in the cache. Hence, performance-wise this does not make a difference. So this might be something worth considering again?

Reply:
Thanks for sharing. Nice first level caching! Did not know. 
However, it feels a bit odd passing the repository through the app. It would also require us to add more logic in the tags provider. So I guess attaching the tenantId from the controller to the request seems like the best solution. 
I don't fully understand what you mean by "it is too late", since the tenantId is being attached to the metrics, thus, it works. 
Will extract the logic into a function with Urs, though :)